sphinx/roles.py

```diff
@@ -68,19 +68,14 @@ class XRefRole:
     * Subclassing and overwriting `process_link()` and/or `result_nodes()`.
     """
 
-    nodeclass = addnodes.pending_xref  # type: Type[nodes.Node]
-    innernodeclass = nodes.literal
```

Comment:
At this moment, subclasses can override the default nodeclass and innernodeclass by overriding the class attribute. But this change breaks it. The technique is not used in Sphinx-core. But I can't say about 3rd party extensions.
I don't know it is promised interface, but this must be a breaking change. So we have to provide migration pass and deprecation warnings.

IMO, it seems this is styling change. So the migration is a bit tough work compared with benefit.

Reply:
> At this moment, subclasses can override the default nodeclass and innernodeclass by overriding the class attribute.

I have missed this point, so I will revert the initialization code.